src/sentry/search/events/fields.py

```diff
@@ -1312,114 +1335,80 @@ def is_accessible(self, acl=None):
         ),
         Function(
             "apdex",
-            required_args=[NumberRange("satisfaction", 0, None)],
-            transform="apdex(duration, {satisfaction:g})",
-            default_result_type="number",
-        ),
-        Function(
-            "apdex_new",
-            transform="""
-                apdex(
-                    multiIf(
-                        equals(
-                            tupleElement(project_threshold_config, 1),
-                            'lcp'
-                        ),
-                        if(
-                            has(measurements.key, 'lcp'),
-                            arrayElement(measurements.value, indexOf(measurements.key, 'lcp')),
-                            NULL
+            optional_args=[with_default(0, NumberRange("satisfaction", 0, None))],
```

Comment:
Why does the `satisfaction` arg require a default of `0`?

Reply:
All optional args currently require a default or have a way to get default and I don't think I should change the mehaviour. But basically the "conditional function" works by evaluating the first element of the tuple (which in our case will either be the apdex theshold or 0) and if it evaluetes to true (ie there is a thershold specified), it will use the legacy apdex, otherwise it will use the new project threshold.

Reply:
Should the default be `None` instead of `0`? While `0` is an invalid satisfaction arg (at least not useful), it's not immediately obvious from looking at this that `0` means use the new apdex function.

Reply:
Oh lol yes good call, for some reason I thought it could be just a numeric default, didn't even try None

Reply:
I just ended up adding a new type c0d0372e4776e2c18d7379bbe9899043f775a77d

Reply:
renamed to `NullableNumberRange` in the subsequent commit